src/modules/Writers.jl

```diff
@@ -59,13 +59,91 @@ function render(io::IO, mime::MIME"text/plain", node::Expanders.DocsNodes, page,
     end
 end
 
+"""
+Converts the function argument tuple `(name, type)` into a string.
+
+The tuple comes from the second return element of the `Base.arg_decl_parts(::Method)`
+and it seems they are always both `::String` (`::ASCIIString` in 0.4).
+It also appears that if the type is not declared for the method, `arg_decl_parts`
+returns an empty string.
+
+The returned string is `name::type` or just `name`, if the type is not declared.
+"""
+function join_decl(decl)
+    n, t = decl
+    isempty(t) ? n : "$(n)::$(t)"
+end
+
 function render(io::IO, mime::MIME"text/plain", node::Expanders.DocsNode, page, doc)
     # Docstring header based on the name of the binding and it's category.
     anchor = "<a id='$(node.anchor.id)' href='#$(node.anchor.id)'>#</a>"
     header = "**`$(node.object.binding)`** &mdash; *$(Utilities.doccat(node.object))*."
     println(io, anchor, "\n", header, "\n\n")
     # Body. May contain several concatenated docstrings.
     renderdoc(io, mime, node.docstr, page, doc)
+
+    # Table of methods.
+    # If DocsNode.methods is nulled, then we assume that we should not render a
+    # table. However, if the list of methods is there but has no elements then
+    # we output an appropriate note saying that the name has no methods associated
+    # with it.
+    Utilities.unwrap(node.methods) do ms
+        println(io, "<p><strong>Methods</strong></p>")
+
+        if isempty(ms)
+            println(io, "<p><emph>This name has no methods.</emph></p>")
```

Comment:
Since we've got access here to whether it's a function or a type we should include the relevant one. Though it will only ever be empty for a `Function` object defined with

```julia
function func end
```

so probably just put `This function has no methods.` and call it done I think.

Regarding constructor/method, lets just leave as you've written it. If anyone comes along to complain about it we can deal with it then.